From 8dc42793344c479e89a130235da8a26ee3684079 Mon Sep 17 00:00:00 2001
From: Yong Zhang <yong.zhang0@gmail.com>
Date: Thu, 22 Sep 2011 16:59:15 +0800
Subject: TTY: irq: Remove IRQF_DISABLED

Since commit [e58aa3d2: genirq: Run irq handlers with interrupts disabled],
We run all interrupt handlers with interrupts disabled
and we even check and yell when an interrupt handler
returns with interrupts enabled (see commit [b738a50a:
genirq: Warn when handler enables interrupts]).

So now this flag is a NOOP and can be removed.

Signed-off-by: Yong Zhang <yong.zhang0@gmail.com>
Acked-by: Tobias Klauser <tklauser@distanz.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
(cherry picked from commit 9cfb5c05fee914cc65d4706801f6bc424082b5f5)

Conflicts:

	drivers/tty/serial/bfin_uart.c
	drivers/tty/amiserial.c
	drivers/tty/cyclades.c
	drivers/tty/hvc/hvc_irq.c
	drivers/tty/hvc/hvcs.c
	drivers/tty/hvc/hvsi.c
	drivers/tty/isicom.c
	drivers/tty/serial/68328serial.c
	drivers/tty/serial/altera_jtaguart.c
	drivers/tty/serial/altera_uart.c
	drivers/tty/serial/bfin_sport_uart.c
	drivers/tty/serial/crisv10.c
	drivers/tty/serial/icom.c
	drivers/tty/serial/lantiq.c
	drivers/tty/serial/mcf.c
	drivers/tty/serial/mpc52xx_uart.c
	drivers/tty/serial/serial_ks8695.c
	drivers/tty/serial/sn_console.c

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 drivers/tty/serial/sh-sci.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/sh-sci.c b/drivers/tty/serial/sh-sci.c
index 16914a1..a385564f 100644
--- a/drivers/tty/serial/sh-sci.c
+++ b/drivers/tty/serial/sh-sci.c
@@ -1972,7 +1972,7 @@ static int __devinit sci_init_single(struct platform_device *dev,
 	 * For the muxed case there's nothing more to do.
 	 */
 	port->irq		= p->irqs[SCIx_RXI_IRQ];
-	port->irqflags		= IRQF_DISABLED;
+	port->irqflags		= 0;
 
 	port->serial_in		= sci_serial_in;
 	port->serial_out	= sci_serial_out;
-- 
1.7.10

